[shuenj] iP

shuenj opened the pull request:
# ReminderDude

> I will deal with that tomorrow.

ReminderDude is a chatbot that reminds you of the tasks you need to do. 😄 
ReminderDude is:
- Simple
- Fast
- Easy to use

## The latest updates allow users to:
1. Check your daily schedule
2. Find your tasks

## To use it, follow the steps below:
- [x] *Deposit* $5 per task.
- [ ] Get **$4.50 BACK** every time a tasks is completed!
- [ ] ~~Forget your task~~ Pay 50 cents more to check your tasks!

## Try coding ReminderDude yourself with the code below  :thumbsup:: 
   ```java
   public class HelloWorld {
    public static void main(String[] args) {
        System.out.println("Hello, World!");
    }
  }
```
## Can't do it yourself? Buy ReminderDude now or learn SE at [CS2103 Website](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html)!!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

shuenj committed:
Add initial skeletal version

shuenj committed:
Add echo feature

shuenj committed:
add list

shuenj committed:
add Task.java

shuenj committed:
add markAsUndone function

shuenj committed:
add mark as done

shuenj committed:
add Deadline.java

shuenj committed:
add Todo.java

shuenj committed:
add Event.java

shuenj committed:
add ToDos, Events, Deadlines to Duke

shuenj committed:
add test case and output

shuenj committed:
add TaskManager class

shuenj committed:
move println to Task object

shuenj committed:
move print to TaskManager

shuenj committed:
add error handling

shuenj committed:
add delete feature

shuenj committed:
add delete feature

shuenj committed:
use enum for cases

shuenj committed:
Add Storage class

shuenj committed:
Add save and load to file for all tasks

shuenj committed:
Handle file does not exist or has content

shuenj committed:
Handle corrupt file

shuenj committed:
Merge branch 'branch-Level-7'

shuenj committed:
Store dates in Deadline and Event as LocalDateTime objects

shuenj committed:
Add command to print schedule on specific date

shuenj committed:
Merge branch 'branch-Level-8'

shuenj committed:
Add Ui class

shuenj committed:
Add Add, Delete, Exit, Invalid, List, Schedule Commands

shuenj committed:
Rename TaskManager to TaskList

shuenj committed:
Refactor Storage and Parser

shuenj committed:
Refactor MarkCommand and UnMarkCommand

shuenj committed:
Refactor Save and Load Commands

shuenj committed:
Clean commented out code

shuenj committed:
Merge branch 'branch-A-MoreOOP'

shuenj committed:
Organize into packages

shuenj committed:
Merge branch 'branch-A-Packages'

shuenj committed:
Merge remote-tracking branch 'origin/add-gradle-support'

shuenj committed:
Add Gradle support

shuenj committed:
Merge branch 'branch-A-Gradle'

shuenj committed:
Add unit test for Deadline and ScheduleCommand

shuenj committed:
Merge branch 'branch-A-JUnit'

shuenj committed:
Add Javadocs for command classes

shuenj committed:
Add Javadocs for exception class

shuenj committed:
Add Javadocs for io classes

shuenj committed:
Add Javadocs for Task classes

shuenj committed:
Add Javadocs for Ui classe

shuenj committed:
Add Javadocs for Duke class

shuenj committed:
Comply with coding standard with checkstyle

shuenj committed:
Comply with coding standards and checkstyle

shuenj committed:
Add find command

shuenj committed:
Import FindCommand class

shuenj committed:
Merge branch 'branch-Level-9'

shuenj committed:
Merge branch 'branch-A-CodingStandard'

shuenj committed:
Resolved merge conflicts

shuenj committed:
Fixed merge conflicts in tasks

shuenj committed:
Merge conflicts in tasks

shuenj committed:
Add comments for FindCommand using checkstyle

shuenj committed:
Merge branch 'branch-A-CheckStyle'

shuenj committed:
Update build.gradle

shuenj committed:
Add gui package

shuenj committed:
Add GUI and fxml

shuenj committed:
Merge branch 'branch-Level-10'

shuenj committed:
Add varargs to ui getOutput method

shuenj committed:
Experiment with gui

shuenj committed:
Merge branch 'branch-A-Varargs'

lshaoqin commented on src/main/java/commands/AddCommand.java:
```diff
@@ -0,0 +1,84 @@
+     * @param storage The Storage object that handles the saving and loading of tasks
+     * @throws DukeException If the command is invalid
+     */
+    public void execute(TaskList taskList, Ui ui, Storage storage) throws DukeException {
```
Good practice to annotate `@Override` here and in your other classes

lshaoqin commented on src/main/java/tasks/EventTask.java:
```diff
@@ -0,0 +1,92 @@
+     * Convert the event task to a string to be saved in a file.
+     * @return String
+     */
+    public String toFileString() {
```
As above, it is good to annotate these with `@Override`

lshaoqin commented on src/test/java/tasks/DeadlineTaskTest.java:
```diff
@@ -0,0 +1,29 @@
+
+public class DeadlineTaskTest {
+    @Test
+    public void fromFileStringTest() {
```
perhaps these could be better named `readFromFileTest` and `writeToFileTest`?

lshaoqin commented on src/main/java/commands/AddCommand.java:
```diff
@@ -0,0 +1,84 @@
+/**
+ * Represents a command to add a task to the list of tasks.
+ */
+public class AddCommand extends Command {
```
Refactoring this as `AddTaskCommand` might improve readability slightly.

lshaoqin reviewed (commented):
Great job! I pointed out a few places where you could add the `@Override` annotation for safety, and one or two places where slightly better names could improve readability, but very well formatted and clean code overall.

yanghengtang commented on src/main/java/commands/AddCommand.java:
```diff
@@ -0,0 +1,84 @@
+package commands;
+import exceptions.DukeException;
```
Adding a line between `package` statement and `import` statements can help browsing through the list easier.

yanghengtang commented on src/main/java/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,39 @@
+     * @param storage The Storage object that handles the saving and loading of tasks
+     * @throws DukeException If the command is invalid
+     */
+    public void execute(TaskList taskList, Ui ui, Storage storage) throws DukeException {
```
Adding `@Override` before the method would be a good in all your subclasses of Command

yanghengtang commented on src/main/java/commands/MarkCommand.java:
```diff
@@ -0,0 +1,38 @@
+     * Constructor for MarkCommand.
+     * @param fullCommand
+     */
+    public MarkCommand(String fullCommand) {
```
Is there a reason why `fullCommand` is used as the parameter name as opposed to `command` in similar subclasses of `Command` class?

shuenj replied:
good catch, thanks!

yanghengtang commented on src/main/java/tasks/DeadlineTask.java:
```diff
@@ -0,0 +1,87 @@
+public class DeadlineTask extends Task {
+
+    /** Datetime of deadline of task */
+    protected LocalDateTime by;
```
For better clarity, `by` could be renamed to `byDate`.

yanghengtang commented on src/main/java/tasks/EventTask.java:
```diff
@@ -0,0 +1,92 @@
+
+    /**
+     * Constructor for EventTask.
+     * @param description
```
Missing description for each parameter's JavaDoc

yanghengtang reviewed (commented):
Good work on your chatbot so far

shuenj committed:
Add assert in deleteCommand

shuenj committed:
Add assert to storage to ensure file exists

shuenj committed:
Add assert to parser to ensure command array is not empty

shuenj committed:
Add getter for tasklist

shuenj committed:
Use streams to write file

shuenj committed:
Merge pull request #1 from shuenj/branch-A-Streams

Branch a streams

shuenj committed:
Merge branch 'master' into branch-A-Assertions

shuenj committed:
Merge pull request #2 from shuenj/branch-A-Assertions

Branch a assertions

shuenj committed:
Edits based on peer PR review.

Class method over-writing its base class method lacks @Override annotation.

Safety can be improved by having the annotation.

The annotation has been added to methods in the Tasks and Command classes.

Other edits include:
Change variable names for better readability.
Add line between package and import statements.

shuenj committed:
Edits on Javadocs

The parameters lack a description.

This can be confusing as there are variables with similar names but different type.

Added description to parameters of methods in Javadocs.

shuenj committed:
Checkstyle edits

shuenj committed:
Merge pull request #3 from shuenj/branch-A-CodeQuality

Branch a code quality

shuenj committed:
Add snooze Command and ui

shuenj committed:
checkstyle changes

shuenj committed:
Merge branch 'branch-B-Snooze'

shuenj committed:
Add user guide

shuenj committed:
Merge branch 'branch-A-UserGuide'

shuenj committed:
Add Ui.png

shuenj committed:
Add line between description and parameter section

shuenj committed:
Add lines to user guide

shuenj committed:
Merge branch 'branch-BotIssues'
